ecdsa/src/sp1.rs

```diff
@@ -8,62 +8,69 @@ use elliptic_curve::{
 };
 
 use elliptic_curve::Field;
-use sp1_lib::io::{self, FD_ECRECOVER_HOOK};
+use sp1_lib::io::{self, K1_ECRECOVER_HOOK, R1_ECRECOVER_HOOK};
 use sp1_lib::unconstrained;
 use sp1_lib::{
-    secp256k1::Secp256k1Point, syscall_secp256k1_decompress, utils::AffinePoint as Sp1AffinePoint,
+    secp256k1::Secp256k1Point, secp256r1::Secp256r1Point, syscall_secp256k1_decompress, syscall_secp256r1_decompress, utils::AffinePoint as Sp1AffinePoint,
 };
 
 use crate::{hazmat::bits2field, Signature, SignatureSize, VerifyingKey};
 
+#[derive(Debug, Clone, Copy)]
+pub enum Secp256Curve {
+    K1, // secp256k1
+    R1, // secp256r1
+}
+
 #[cfg(feature = "verifying")]
 impl<C> VerifyingKey<C>
 where
     C: PrimeCurve + CurveArithmetic,
-    AffinePoint<C>: DecompressPoint<C> + FromEncodedPoint<C> + ToEncodedPoint<C>,
-    FieldBytesSize<C>: sec1::ModulusSize,
     SignatureSize<C>: ArrayLength<u8>,
 {
     /// Recover a [`VerifyingKey`] from the given `prehash` of a message, the
     /// signature over that prehashed message, and a [`RecoveryId`].
     ///
     /// This function leverages SP1 syscalls for secp256k1 to accelerate public key recovery
     /// in the zkVM. Verifies the signature against the recovered public key to ensure correctness.
-    pub fn recover_from_prehash_secp256k1(
+    pub fn recover_from_prehash_secp256(
         prehash: &[u8],
         signature: &Signature<C>,
         recovery_id: RecoveryId,
-    ) -> Result<Self> {
+        curve: Secp256Curve,
+    ) -> Result<[u8; 65]> { //Result<Self>
```

Comment:
nit: Remove Result<Self>

Also, the description of the function should include that it returns the uncompressed public key bytes in BE/LE format